File: Python/challenge29.py
import sys

def remove_duplicates(s):
    """Returns a string s with duplicated elements removed."""
    output = []
    for elem in s.strip().split(','):
        if elem not in output:
            output.append(elem)

    return ','.join(output)

test_cases = open(sys.argv[1], 'r')

for test in test_cases:
    print remove_duplicates(test)

test_cases.close()

File: Ruby/challenge29.rb
def remove_duplicates(s)
  # Remove duplicates from a string
  output = []
  s.strip.split(',').each { |e| output << e if not output.include?(e) }
  return output.join(',')
end

File.open(ARGV[0]).each_line do |line|
  puts remove_duplicates(line)
end
